src/transformers/models/whisper/tokenization_whisper.py

```diff
@@ -1015,6 +1052,149 @@ def _find_longest_common_sequence(sequences):
         left_sequence = right_sequence[right_mid:]
         left_length = len(left_sequence)
 
+        if token_timestamp_sequences:
+            total_token_timestamp_sequence.extend(left_token_timestamp_sequence[:left_mid])
+            left_token_timestamp_sequence = token_timestamp_sequences[seq_idx + 1][right_mid:]
+
     total_sequence.extend(left_sequence)
 
-    return total_sequence
+    if token_timestamp_sequences is None:
+        return total_sequence
+
+    if len(token_timestamp_sequences) > 0:
+        total_token_timestamp_sequence.extend(left_token_timestamp_sequence)
+        return total_sequence, total_token_timestamp_sequence
+    else:
+        return total_sequence, []
+
+
+def _collate_word_timestamps(tokenizer, tokens, token_timestamps, language):
```

Comment:
Nice! These functions LGTM, clean top-level integration into our current `_decode_asr` method IMO, with the complexity tucked away in these private functions